### DISCUSSION TITLE
Handle cross-domain authentication / live preview

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9452] [Ideas]
> Created by: Timothy Kelty (timkelty@gmail.com) on 2015/04/02 12:33:05 +0000
> Votes at time of UserVoice import: 2

-----------------------------------

If your locales have different TLDs, there isn't a good way to keep users logged in across domains. A side-effect of this is live preview doesn't work.

http://craftcms.stackexchange.com/questions/7187/multi-locale-multi-domain-live-preview-failing/12268#12268

http://craftcms.stackexchange.com/questions/14963/live-preview-in-different-languages-on-different-top-level-domains

**Comment by timkelty** [comment]
Will this feature also provide a method for cross-tld logins?

**Comment by brandonkelly** [comment]
For that you would need to use OAuth. We do have an OAuth server plugin that we use for Craftnet/the Plugin Store: https://github.com/pixelandtonic/oauth-server

**Comment by timkelty** [comment]
@brandonkelly hmm that link is broken - is it private?

**Comment by brandonkelly** [comment]
hah sorry, private repo. It’s specific to our needs for now anyway, but we do plan on eventually making it public.

**Comment by brandonkelly** [comment]
Worth mentioning here that Live Preview will be token-based in 3.1 (1ab6b6e6e3cd5a13c949d113c8367e7faf930ee4), so will no longer be affected by this issue. But I’ll leave the FR open as it’s not strictly related to LP.

**Comment by lassemt** [comment]
If cross-domain authentication were to be implemented, would this be through an OAuth server?

**Comment by brandonkelly** [comment]
@lassemt Probably, or something like that.
